app/scripts/controllers/pipelines.js

```diff
@@ -0,0 +1,140 @@
+'use strict';
+
+/**
+ * @ngdoc function
+ * @name openshiftConsole.controller:PipelinesController
+ * @description
+ * # ProjectController
+ * Controller of the openshiftConsole
+ */
+angular.module('openshiftConsole')
+  .controller('PipelinesController', function ($filter,
+                                               $routeParams,
+                                               $scope,
+                                               AlertMessageService,
+                                               BuildsService,
+                                               DataService,
+                                               Logger,
+                                               ProjectsService) {
+    $scope.projectName = $routeParams.project;
+    $scope.alerts = $scope.alerts || {};
+    $scope.buildConfigs = {};
+    $scope.statsByConfig = {};
+
+    // get and clear any alerts
+    AlertMessageService.getAlerts().forEach(function(alert) {
+      $scope.alerts[alert.name] = alert.data;
+    });
+    AlertMessageService.clearAlerts();
+
+    var watches = [];
+
+    ProjectsService
+      .get($routeParams.project)
+      .then(_.spread(function(project, context) {
+        $scope.project = project;
+
+        var builds, statsByConfig = {};
+
+        var buildConfigForBuild = $filter('buildConfigForBuild');
+        var isIncomplete = $filter('isIncompleteBuild');
+        var isPipeline = $filter('isJenkinsPipelineStrategy');
+        var isNewer = $filter('isNewerResource');
+
+        var updateStats = function(buildConfigName, build) {
+          if (!statsByConfig[buildConfigName]) {
+            statsByConfig[buildConfigName] = {
+              count: 0,
+              // total duration of all builds for calculating averages
+              totalDuration: 0
+            };
+          }
+          statsByConfig[buildConfigName].count++;
+          statsByConfig[buildConfigName].totalDuration += BuildsService.getDuration(build);
```

Comment:
I can do that. Thinking of filtering out incomplete builds from avg duration as well.